### PR TITLE
Bump go attestation

### PR DESCRIPTION
**Pull request by pawalt**
This PR:
- bumps us to go-attestation version 0.1.2 and fixes incompatibilities in the upgrade
- fixes bug where we were hashing the whole EK cert instead of just the public key
- modifies `get_tpm_pubhash` retrieval method to be consistent with rest of project

### CODE REVIEW

File: cmd/server/tpm_attestor/tpm_attestor.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- ** Copyright 2019 Bloomberg Finance L.P.
+** Copyright 2019 Bloomberg Finance L.P.
```
Comment (twexler) [inline cmd/server/tpm_attestor/tpm_attestor.go]: accidental?

File: go.mod [hunk 2]
```diff
@@ -18,14 +15,13 @@ require (
 	github.com/mitchellh/go-testing-interface v1.0.0 // indirect
 	github.com/prometheus/client_golang v1.1.0 // indirect
 	github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4 // indirect
+	github.com/prometheus/common v0.6.0
```
Comment (twexler) [inline go.mod]: Wrong `common`?

File: tools/get_tpm_pubhash/get_tpm_pubhash.go [hunk 3]
```diff
@@ -17,33 +17,42 @@
...
-	"flag"
 	"fmt"
 	"log"
 
 	"github.com/bloomberg/spire-tpm-plugin/pkg/common"
-)
-
-var (
-	tpmPath = flag.String("tpm-path", "/dev/tpmrm0", "location of tpm device")
+	"github.com/google/certificate-transparency-go/x509"
```
Comment (twexler) [inline tools/get_tpm_pubhash/get_tpm_pubhash.go:24]: wrong x509 package?

File: cmd/server/tpm_attestor/tpm_attestor.go [hunk 4]
```diff
@@ -12,26 +12,22 @@
...
 	"errors"
 	"fmt"
-	"log"
-
-	"github.com/google/certificate-transparency-go/x509"
+	"io/ioutil"
+	"path/filepath"
 
 	"github.com/bloomberg/spire-tpm-plugin/pkg/common"
+	"github.com/google/certificate-transparency-go/x509"
```
Comment (twexler) [inline cmd/server/tpm_attestor/tpm_attestor.go]: same here, did you mean `encoding/x509`?

**Review by twexler — CHANGES_REQUESTED** [review]
A few things

File: go.mod (hunk 2 above)
Comment (pawalt) [reply go.mod]: Yep good catch

File: cmd/server/tpm_attestor/tpm_attestor.go (hunk 1 above)
Comment (pawalt) [reply cmd/server/tpm_attestor/tpm_attestor.go]: Yep no idea how this hapened

File: tools/get_tpm_pubhash/get_tpm_pubhash.go (hunk 3 above)
Comment (pawalt) [reply tools/get_tpm_pubhash/get_tpm_pubhash.go:24]: Google mantains their own fork of `crypto/x509` that they use for TPMs. go-attestation uses this fork, so I have to as well.

File: cmd/server/tpm_attestor/tpm_attestor.go (hunk 4 above)
Comment (pawalt) [reply cmd/server/tpm_attestor/tpm_attestor.go]: see above

File: tools/get_tpm_pubhash/get_tpm_pubhash.go (hunk 3 above)
Comment (twexler) [reply tools/get_tpm_pubhash/get_tpm_pubhash.go:24]: hmmph.

**Review by twexler — APPROVED** [review]
lgtm

**Pull request merged**